### PR TITLE
HW-51227: Core Paypal Account

### PR DESCRIPTION
**Pull request by peter-joseph**

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/model/PayPalAccount.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+ * EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN
+ * THE SOFTWARE.
```
[the remark this answers is not in the document]
Comment (vshcherbyna-epam) [reply core/src/main/java/com/hyperwallet/android/model/PayPalAccount.java]: replaced

File: core/src/main/java/com/hyperwallet/android/model/PayPalAccount.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        public Builder transferMethodCurrency(@NonNull String transferMethodCurrency) {
+            mFields.put(TransferMethodFields.TRANSFER_METHOD_CURRENCY, transferMethodCurrency);
+            return this;
+        }
+
+        public PayPalAccount build() {
+            return new PayPalAccount(mFields);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (vshcherbyna-epam) [reply core/src/main/java/com/hyperwallet/android/model/PayPalAccount.java]: added

File: core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletCreatePayPalAccountTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Test
+    public void testCreatePayPalAccount_withSuccess() throws InterruptedException {
+
+        String responseBody = mExternalResourceManager.getResourceContent("paypal_account_response.json");
+        mServer.mockResponse().withHttpResponseCode(HttpURLConnection.HTTP_CREATED).withBody(responseBody).mock();
+
+        final PayPalAccount.Builder builder = new PayPalAccount
+                .Builder("US", "USD", "jsmith@paypal.com")
+                .token("trm-ac5727ac-8fe7-42fb-b69d-977ebdd7b48b");
```
Comment (skoong) [inline core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletCreatePayPalAccountTest.java]: Remove token here please

File: core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletCreatePayPalAccountTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Test
+    public void testCreatePayPalAccount_withValidationError() throws InterruptedException {
+        String responseBody = mExternalResourceManager.getResourceContentError("transfer_method_error_response.json");
+        mServer.mockResponse().withHttpResponseCode(HttpURLConnection.HTTP_BAD_REQUEST).withBody(responseBody).mock();
+
+        final PayPalAccount payPalAccount = new PayPalAccount
+                .Builder("", "USD", "jsmith@paypal.com")
+                .token("trm-ac5727ac-8fe7-42fb-b69d-977ebdd7b48b")
```
Comment (skoong) [inline core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletCreatePayPalAccountTest.java]: Also here as well

File: core/src/test/resources/errors/paypal_accounts_error_response.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "errors": [
```
Comment (skoong) [inline core/src/test/resources/errors/paypal_accounts_error_response.json]: Please reuse system_error_response.json instead of adding a new JSON file.

**Pull request merged**